app/vmctl/main.go

```diff
@@ -212,12 +212,14 @@ func main() {
 							Addr:     strings.Trim(c.String(vmNativeSrcAddr), "/"),
 							User:     c.String(vmNativeSrcUser),
 							Password: c.String(vmNativeSrcPassword),
+							Headers:  strings.Split(vmNativeSrcHeaders, "^^"),
```

Comment:
Hi @baconmania ! Thank you for the improvement. This will not work. You should read the flag by cli function `c.String(vmNativeSrcHeaders)` and after checking, if the header is empty or not, split it by provided delimiter Please take a look into remote read realization, there is similar logic implemented.